### ISSUE TITLE
Add a new function to try to get the specific database backup.

### ISSUE THREAD

**Issue by GuChad369** [pgmoneta/pgmoneta_ext#18]
We are using the `BASE_BACKUP` protocol in the core project, so we can't specify which database(s) to back up.

We are trying to fix this problem by using an extension and making it work like `pg_dump`.

**Comment by GuChad369** [comment]
For the next step, Do we need a function to get the backup using the extension by OID and return it in a script similar to `pg_dump`? 

I think this will be complex, so I will start by learning how `pg_dump` works to figure out how to achieve this.

**Comment by GuChad369** [comment]
`pg_dump` returns a script file that contains all the PostgreSQL commands needed to perform the backup. 

this is one approach used by `pg_dump`.

**Comment by jesperpedersen** [comment]
First, we need to figure out all the commands in the script and how to use them in the extension such that it can send a TAR file to pgmoneta/core.

**But**, we have to figure out if it is worth the work. Can you write something up in order to see how much work it would be ?

**Comment by GuChad369** [comment]
I meant that there is an option for `pg_dump` to perform a backup like this:

```console
pg_dump mydb > db.sql
```

This command will return all the SQL commands needed to create the database and insert data, allowing the user to achieve a backup.

I suppose we need to implement a function `pgmoneta_ext_backup_script(OID)` that can return all the SQL needed to perform the backup based on the passed-in OID.

I will start by studying all the functions of `pg_dump`.

**Comment by GuChad369** [comment]
I wonder if we can call `pg_dump` using the extension on the server side and then send the result to the core. This way, we don't need to implement the functions of `pg_dump`; we can just call it and handle its result. By doing this, we can achieve all the functions of `pg_dump`.

**Comment by jesperpedersen** [comment]
Take a look at the `pg_dump -Fc` path

**Comment by jesperpedersen** [comment]
No, we can't depend on PostgreSQL binaries / libraries - the extension has to use the internal API
